Make env variables available in PHP - clearing environment in FPM workers

CodeMuz opened the pull request:
This resolves an issue where the environment variables are not accessible in PHP as Nginx fpm is not releasing them. Therefore Laravel could not see them using getenv() function in either the $_ENV array or the $_SERVER arrays.

CodeMuz committed:
add flag so env variabls are avaiable to php

CodeMuz committed:
set APP_KEY to 32 char len for AES-256-CBC

lucho2d7 commented:
Thank you murray. I just reviewed some code on one of my projects and this is what i am using:
export APP_KEY=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1)
Should we use this in .openshift/action_hooks/deploy instead of your proposal? I don't remember why, but I think the OPENSHIFT_SECRET_TOKEN might contain problematic characters.

CodeMuz commented:
Hi Lucho, I was experimenting with keys earlier and the OPENSHIFT_SECRET_TOKEN variable has valid characters but it's length is an issue for Laravel. (needs 32 chars)

The OPENSHIFT_SECRET_TOKEN appears to be generated using something similar to yours, i.e

`OPENSHIFT_SECRET_TOKEN=$(cat /dev/urandom | LC_CTYPE=C tr -dc _A-Z-a-z-0-9 | fold -w 128 | head -n 1)`

And this then requires slicing down to size. I'll continue looking for more documentation on this key generation. I think for consistency with the build it makes to use:

`export APP_KEY=${OPENSHIFT_SECRET_TOKEN:0:32}`

The downside is it's a dependency that could theoretically change, so I see the benefit of using a more reliable source like yours.

I'll leave it to you as they both seem ok solutions, I'm happy to amend the pull request with your change, also then the file .openshift/.env could have it's note updated on line 14:
`# APP_KEY=(set automatiically using OPENSHIFT_SECRET_TOKEN, do not modify)`

Ideally we could use the output from artisan ($ php artisan key:generate) during deploy
Or something very similiar 
 see: function generateRandomKey() line 61-71
[KeyGenerateCommand.php](https://github.com/laravel/framework/blob/5.2/src/Illuminate/Foundation/Console/KeyGenerateCommand.php)
in bash or with php : `key = 'base64:'.base64_encode(rand(32));`
Assuming the deploy script gets executed after composer has installed all the vendor dependencies..

lucho2d7 commented:
What about the underscore? Did you try using a string with underscore as APP_KEY ? what about dash?

CodeMuz committed:
adding deploy hook to install composer dependencies

CodeMuz committed:
generate key using laravel artisan

CodeMuz committed:
showing variables in php

CodeMuz committed:
using artisan for pre restart hook

CodeMuz commented:
Hi Lucho, you're completely right the underscore and dash are not accepted! I don't know how I convinced myself it was ok!! Sorry!

To resolve this your solution looks good or to be in flavour with _Laravel_ applications I've added a call to artisan which generates keys of this format:
`base64:tRRHaLa6W2zjBhvZCKzyk0hMX/0jlSJ/kYCAfR4wYo4=`

It needs to run composer just before in action_hooks/deploy. I've updated this branch, what do you think?
